src/transformers/modeling_longformer.py

```diff
@@ -707,3 +707,126 @@ def forward(
+            Position outside of the sequence are not taken into account for computing the loss.
+        end_positions (:obj:`torch.LongTensor` of shape :obj:`(batch_size,)`, `optional`, defaults to :obj:`None`):
+            Labels for position (index) of the end of the labelled span for computing the token classification loss.
+            Positions are clamped to the length of the sequence (`sequence_length`).
+            Position outside of the sequence are not taken into account for computing the loss.
+    Returns:
+        :obj:`tuple(torch.FloatTensor)` comprising various elements depending on the configuration (:class:`~transformers.RobertaConfig`) and inputs:
+        loss (:obj:`torch.FloatTensor` of shape :obj:`(1,)`, `optional`, returned when :obj:`labels` is provided):
+            Total span extraction loss is the sum of a Cross-Entropy for the start and end positions.
+        start_scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length,)`):
+            Span-start scores (before SoftMax).
+        end_scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length,)`):
+            Span-end scores (before SoftMax).
+        hidden_states (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``config.output_hidden_states=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for the output of the embeddings + one for the output of each layer)
+            of shape :obj:`(batch_size, sequence_length, hidden_size)`.
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        attentions (:obj:`tuple(torch.FloatTensor)`, `optional`, returned when ``config.output_attentions=True``):
+            Tuple of :obj:`torch.FloatTensor` (one for each layer) of shape
+            :obj:`(batch_size, num_heads, sequence_length, sequence_length)`.
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+    Examples::
+        from transformers import LongformerTokenizer, LongformerForQuestionAnswering
+        import torch
+        tokenizer = LongformerTokenizer.from_pretrained(longformer-base-4096')
+        model = LongformerForQuestionAnswering.from_pretrained(longformer-base-4096')
+        question, text = "Who was Jim Henson?", "Jim Henson was a nice puppet"
+        encoding = tokenizer.encode_plus(question, text)
+        input_ids, attention_mask = encoding["input_ids"], encoding["attention_mask"]
+        start_scores, end_scores = model(torch.tensor([input_ids]), attention_mask=attention_mask)
+        all_tokens = tokenizer.convert_ids_to_tokens(input_ids)
+        answer = ' '.join(all_tokens[torch.argmax(start_scores) : torch.argmax(end_scores)+1])
+        """
+
+        # Compute attention mask if not provided,
+        # set global attention on question tokens 
+        if attention_mask is None:
```

Comment:
Thinking more about this, I think we need to also take care of the padding. We have one of three options
- moving this into the tokenizer as @patrickvonplaten  suggested, but it might get complicated because global attention mask is different for different tasks
- update the `attention_mask` here with zeros in locations corresponding to padding tokens, as you @patil-suraj, did earlier
- Change the `if attention_mask is None:` to 
```if attention_mask is None or no_global_attention_in_attention_mask:
    global_attention_mask = compute_global_attention_mask  # same code you have
    if  attention_mask is None:
        attention_mask = global_attention_mask
    else:
       attention_mask =  combine(global_attention_mask, attention_mask)
```

Reply:
Yes, the third options seems better for now, I will go ahead with it.

Reply:
@ibeltagy I think we still need to take care of padding in the third option as well when `attention_mask` is not passed

Reply:
@ibeltagy - agree that option 3 is the best for now. 

@patil-suraj you don't need to do the padding here.

1) `attention_mask` is `None` => we initialize all tokens with 1 (local attention) and set some tokens to 2 (global attention) according to the function `compute_global_attention_mask`. This means all tokens will be attended (no token has 0 in its `attention_mask` ) as they should be if no `attention_mask` is defined.

2) `attention_mask` is not `None` => merge/ combine `attention_mask` with `global_attention_mask`. Merging / combining should actually just be:
```python
attention_mask = global_attention_mask * attention_mask
```
=>  The idea is: 1) if attention_mask has a 0 => `global_attention_mask * attention_mask` will also have a zero at this position. If `global_attention_mask` has a 2 and `attention_mask` has a 1, `global_attention_mask * attention_mask` will have a 2 at this position. This is the desired behavior. So no need to write a function for this.

In the longer run we should either:
A) add `global_attention_mask` to the API of the forward function of all Reformer models. I would actually be ok with this.
B) Move this preprocessing to the tokenizer.

But let's do this in another PR. 

Let me know if you have questions!